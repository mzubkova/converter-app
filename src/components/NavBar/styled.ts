import styled from "styled-components";

export const StNavBar = styled.div`
  width: 100%;
  height: 50px;
  margin: 0;
  padding: 0;
  top: 0;
  z-index: 2147483000;
  position: sticky;
  background-color: #38a3a5;
`;

export const StNavBarLink = styled.div`
  position: relative;
  color: #fff;
  margin: 0 20px;
  font-size: 18px;

  &:after {
    background: none repeat scroll 0 0 transparent;
    bottom: -2px;
    content: "";
    display: block;
    height: 1px;
    left: 50%;
    position: absolute;
    background: #fff;
    transition: width 0.3s ease 0s, left 0.3s ease 0s;
    width: 0;
  }
  &:hover:after {
    width: 100%;
    left: 0;
  }
`;

export const StNavBarItems = styled.div`
  display: flex;
  height: 100%;
  align-items: center;

  .active {
    ${StNavBarLink} {
      color: #22577A;

    &:hover:after {
      display: none;
    }
  }
`;
