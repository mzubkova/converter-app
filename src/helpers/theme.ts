import { createTheme } from "@mui/material/styles";

export const theme = createTheme({
  palette: {
    primary: {
      main: "#0971f1",
      contrastText: "#053e85",
    },
    darkGrey: {
      main: "#0f0e0e",
      contrastText: "#053e85",
    },
    pink: {
      main: "#ff5677",
      contrastText: "#053e85",
    },
    blue: {
      main: "#22577a",
      contrastText: "#80ed99",
    },
  },
});
