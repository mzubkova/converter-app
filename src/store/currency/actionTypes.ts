export enum ActionTypes {
  GET_EXCHANGE_RATES = "@@currency/GET_EXCHANGE_RATES",
  GET_EXCHANGE_RATES_SUCCESS = "@@currency/GET_EXCHANGE_RATES_SUCCESS",
  GET_EXCHANGE_RATES_ERROR = "@@currency/GET_EXCHANGE_RATES_ERROR",
}
