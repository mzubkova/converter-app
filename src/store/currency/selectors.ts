import { createSelector } from "reselect";
import { ApplicationState } from "../types";
import * as Types from "./types";

export const getCurrencyState = (
  state: ApplicationState
): Types.ICurrencyStore => state.currency;
export const getExchangeRatesState = createSelector(
  getCurrencyState,
  (currency: Types.ICurrencyStore): Array<any> => currency.exchange_rates
);
export const getIsLoading = createSelector(
  getCurrencyState,
  (currency: Types.ICurrencyStore): boolean => currency.isLoading
);
