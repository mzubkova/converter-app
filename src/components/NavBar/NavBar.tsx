import React from "react";
import { NavLink } from "react-router-dom";
import { StNavBar, StNavBarItems, StNavBarLink } from "./styled";

const NavBar = () => (
  <StNavBar data-testid="nav-bar">
    <StNavBarItems>
      <NavLink
        to="/"
        className={({ isActive }) => (isActive ? "active" : "inactive")}
      >
        <StNavBarLink>Converter</StNavBarLink>
      </NavLink>
      <NavLink to="/currency_list">
        <StNavBarLink>Currency List</StNavBarLink>
      </NavLink>
    </StNavBarItems>
  </StNavBar>
);

export default NavBar;
