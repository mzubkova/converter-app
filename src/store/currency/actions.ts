import { action } from "typesafe-actions";
import { ActionTypes as AT } from "./actionTypes";
import { IExchangeRate } from "./types";

export const getExchangeRates = () => action(AT.GET_EXCHANGE_RATES);
export const getExchangeRatesSuccess = (payload: Array<IExchangeRate>) =>
  action(AT.GET_EXCHANGE_RATES_SUCCESS, payload);
export const getExchangeRatesError = () => action(AT.GET_EXCHANGE_RATES_ERROR);
