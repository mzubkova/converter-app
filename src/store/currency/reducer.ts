import { Reducer } from "redux";
import { ActionTypes as AT } from "./actionTypes";
import * as Types from "./types";

export const initialState: Types.ICurrencyStore = {
  exchange_rates: [],
  isLoading: false,
};

export const currencyReducer: Reducer<
  Types.ICurrencyStore,
  Types.TCurrencyActions
> = (state = initialState, action) => {
  switch (action.type) {
    case AT.GET_EXCHANGE_RATES:
      return {
        ...state,
        isLoading: true,
      };
    case AT.GET_EXCHANGE_RATES_SUCCESS:
      return {
        ...state,
        exchange_rates: action.payload,
        isLoading: false,
      };
    case AT.GET_EXCHANGE_RATES_ERROR:
      return {
        ...state,
        isLoading: false,
      };
    default:
      return state;
  }
};
