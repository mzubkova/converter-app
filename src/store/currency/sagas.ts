import { put, takeEvery, call } from "redux-saga/effects";
import { ActionTypes as AT } from "./actionTypes";
import { SagaIterator } from "@redux-saga/core";
import * as actions from "./actions";
import * as request from "../../api/index";

export function* getExchangeRates(): SagaIterator {
  try {
    const data = yield call(request.getRequest);
    yield put(actions.getExchangeRatesSuccess(data));
  } catch {
    yield put(actions.getExchangeRatesError());
  }
}

export default function* actionWatcher(): SagaIterator {
  yield takeEvery(AT.GET_EXCHANGE_RATES, getExchangeRates);
}
