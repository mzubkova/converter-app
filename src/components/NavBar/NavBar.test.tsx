import * as React from "react";
import NavBar from "../NavBar";
import { shallowRender, mockStore } from "../../../__tests__/helpers";
import { initialState } from "../../store/currency/reducer";
import { screen } from "@testing-library/react";

describe("NavBar", () => {
  let initialStore = {};
  let component: any;
  beforeEach(() => {
    initialStore = mockStore({ currency: initialState });
    component = shallowRender(<NavBar />, initialStore);
  });
  it("should render without crashing", () => {
    expect(component).toMatchSnapshot();
  });
  it("should mount h1", () => {
    expect(screen.getByTestId("nav-bar")).toBeInTheDocument();
  });
});
