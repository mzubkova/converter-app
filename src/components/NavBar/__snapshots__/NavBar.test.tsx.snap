// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styled__StNavBar-sc-abrqqb-0 xCpyq"
        data-testid="nav-bar"
      >
        <div
          class="styled__StNavBarItems-sc-abrqqb-2 kJtHkR"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <div
              class="styled__StNavBarLink-sc-abrqqb-1 eWvXAj"
            >
              Converter
            </div>
          </a>
          <a
            class=""
            href="/currency_list"
          >
            <div
              class="styled__StNavBarLink-sc-abrqqb-1 eWvXAj"
            >
              Currency List
            </div>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styled__StNavBar-sc-abrqqb-0 xCpyq"
      data-testid="nav-bar"
    >
      <div
        class="styled__StNavBarItems-sc-abrqqb-2 kJtHkR"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <div
            class="styled__StNavBarLink-sc-abrqqb-1 eWvXAj"
          >
            Converter
          </div>
        </a>
        <a
          class=""
          href="/currency_list"
        >
          <div
            class="styled__StNavBarLink-sc-abrqqb-1 eWvXAj"
          >
            Currency List
          </div>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
